### ISSUE TITLE
[4.2.0][APICTL] Issue with Externalizing the VCS configuration in APICTL

### ISSUE THREAD

**Issue by hisanhunais** [wso2/api-manager#927]
### Description

After externalizing the VCS configuration by setting the vcs-config-path to a new location, deploying the changes using the apictl vcs deploy will print the following error.
![Screenshot from 2022-11-09 14-54-48](https://user-images.githubusercontent.com/30455603/201580045-940e38db-a703-4778-84de-b65f3196b0fb.png)

According to the documentation [1],
`By setting the above, apictl vcs deploy command will create the vcs_config.yaml if it is not available in the specified path and reuse it for the succeeding commands.`

[1] https://apim.docs.wso2.com/en/latest/install-and-setup/setup/api-controller/advanced-topics/configuring-git-integration/#externalizing-the-vcs-configuration

### Steps to Reproduce

1. Export an API from dev environment
2. Extract it and add it to a git reposidtory
3. Commit it (git add & git commit)
4. apictl vcs init
5. Set the vcs-config-path to a custom location by executing apictl set --vcs-config-path /home/foo/vcs-config.yaml
6. apictl vcs deploy -e prod

Related Documents
1. https://apim.docs.wso2.com/en/latest/install-and-setup/setup/api-controller/advanced-topics/configuring-git-integration/#externalizing-the-vcs-configuration
2. https://apim.docs.wso2.com/en/latest/install-and-setup/setup/api-controller/cicd-using-cli/

### Affected Component

APICTL

### Version

APICTL 4.2.0 M1 testing pack

### Environment Details (with versions)

OS: Ubuntu 20.04 LTS
JDK Version: openjdk version "17.0.5"

### Relevant Log Output

```shell
apictl: Unable to write configuration to file. Reason: open <new_vcs_config_path>: is a directory
```

**Comment by wasuradananjith** [comment]
The above issue was not reproducible for me. The steps that I followed are as follows.

1. Add two environments
    ![image](https://user-images.githubusercontent.com/25246848/203926469-c1beeaaa-0853-45b6-9225-2d6b03ef1d9e.png)
2. Create, Deploy and Publish a sample PizzaShack API in dev environment
    ![image](https://user-images.githubusercontent.com/25246848/203926691-39693d8f-1828-4d5b-8b4c-e34e35be5d9b.png)
3. Export the above API and add it to the Source repo
    ![image](https://user-images.githubusercontent.com/25246848/203927045-a9a899e7-4010-4366-a73f-b150e71d32d3.png)
4. Git init the Source repo and commit the project files.
    ![image](https://user-images.githubusercontent.com/25246848/203927250-eb542356-25fa-437d-9655-2939235a8f1a.png)
5. Execute `apict vcs init`
    ![image](https://user-images.githubusercontent.com/25246848/203927441-caed184b-d0c7-4aa5-8088-1fdac2080fe1.png)
6. Set the vcs-config-path to a custom location `apictl set --vcs-config-path /home/foo/vcs-config.yaml`
    ![image](https://user-images.githubusercontent.com/25246848/203927846-e900c2a3-2250-4be5-b5c4-bb4f67bdf2a7.png)
7. Set source repo path.
    ![image](https://user-images.githubusercontent.com/25246848/203928110-5c7d4665-1287-4316-ba99-e90597a3aa85.png)
8. Execute `apictl vcs deploy -e prod`
   ![image](https://user-images.githubusercontent.com/25246848/203928264-6ba5029b-820a-4e46-bc27-506911be5b8c.png)

@hisanhunais will you be able to confirm whether the steps that I followed are correct?

**Comment by hisanhunais** [comment]
Hi Wasura,

I have made a small mistake in the point 5 of the Steps to Reproduce section. When specifying the location of the custom path, I provided the folder location and did not specify the file name as given in the example block in the documentation [1]. 

In the steps you have mentioned, in point 6, you have specified the file name as well.

In that case, I think the documentation should be updated. If that's the case, let's create a doc issue and solve it.

[1] https://apim.docs.wso2.com/en/latest/install-and-setup/setup/api-controller/advanced-topics/configuring-git-integration/#externalizing-the-vcs-configuration

Thank you,
Hisan
